src/main/java/uk/gov/hmcts/reform/sendletter/logging/InternalDependency.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.sendletter.logging;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to be used against methods to be picked up by AppInsights aspect dependency advice.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface InternalDependency {
+
+    /**
+     * Name of the internal dependency to be tracked as an event in Azure AppInsights.
+     *
+     * @return Name of the event
```

Comment:
event or dependency?

Reply:
The description formally states that **internal dependency** is tracked as an **event**, hence the event name